src/coreclr/src/debug/debug-pal/win/diagnosticsipc.cpp

```diff
@@ -119,6 +119,7 @@ bool IpcStream::DiagnosticsIpc::Listen(ErrorCallback callback)
                 _hPipe = INVALID_HANDLE_VALUE;
                 ::CloseHandle(_oOverlap.hEvent);
                 _oOverlap.hEvent = INVALID_HANDLE_VALUE;
+                memset(&_oOverlap, 0, sizeof(OVERLAPPED)); // clear the overlapped objects state
```

Comment:
This will overwrite _oOverlap.hEvent (currently INVALID_HANDLE_VALUE==-1) with 0, so the check on 249 `if (_oOverlap.hEvent != INVALID_HANDLE_VALUE)` won't do the right thing.

Reply:
Moved the memset to before setting the handle to INVALID_HANDLE_VALUE